javalib/src/main/scala/java/lang/process/UnixProcessGen2.scala

```diff
@@ -76,29 +69,51 @@ private[lang] class UnixProcessGen2 private (
     outfds: Ptr[CInt],
     errfds: Ptr[CInt]
 ) extends UnixProcess() {
+
+  private class CachedExitValue() {
+    final val EXIT_VALUE_UNKNOWN = jl.Integer.MIN_VALUE
+
+    // Non-negative values indicate exit status is known to be that value.
+    val cached = new juc.atomic.AtomicInteger(EXIT_VALUE_UNKNOWN)
```

Comment:
CachedExitValue can extends the AtomicInteger to save some allocation

Reply:
Thank you for the suggestion.

I wanted to centralize all the handling of the 'cached' value in one
place so I created a class.

One could indeed save an instance allocation by having private methods
and a local variable.  Early iterations of this PR did exactly that. It reminded
me of very early Fortran II code.   I was finding that having access to the
variable spread all over the place made it very difficult to tell who as
modifying the variable and when. Having methods makes the handling
much more consistent.  It is harder to have a slightly wrong synchronization
is one (or more) places and miss it.

Reply:
On reading the suggestion several more times, I think I see the essence of it.
// not compiled
```
private class CachedExitValue() extends AtomicInteger(){
    final val EXIT_VALUE_UNKNOWN = jl.Integer.MIN_VALUE

    // Non-negative values indicate exit status is known to be that value.
    val cached = EXIT_VALUE_UNKNOWN

```

That would preserve the class and indeed reduce an allocation at Process creation.

He-Pin is that code somewhat like you suggested?